design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+| 8 | Unable to retrieve Conjur variables | You are using SP in “file” mode | | Error: Failed to provide DAP/Conjur secrets<br><br>SP fails | CSPFK016E | |
+| 9 | Missing volume mounts | You are using SP in “file” mode | One or more volumes cannot be found by Secrets Provider. | Error: Unable to access volume “{value}”. Error: “{value}”. Ensure that the corrrect volumes are defined in the deployment manifest and attached to the Secrets Provider container.<br><br>SP fails to start | | |
+| 10 | File permissions error | You are using SP in “file” mode | Secrets Provider does not have permission to write a secrets file to the specified volume and mount path. | Error: Secrets Provider does not have permission to write the secrets file “{value}”. Please check the volume mount permissions on the Secrets Provider container and try again.<br><br>SP fails | | |
+| 11 | Invalid file path provided | You are using SP in “file” mode and have provided a value for the conjur.org/secret-file-path annotation | The provided file path is not a valid file path. | Error: You have provided an invalid file path “{value}” for the “{value}” secret group.<br><br>SP fails to start | | |
+| 12 | File conflict between two secret groups | You are using SP in "file" mode and have defined multiple secret groups | One or more secret groups have he same ouput file. | Error: You have provided conflicting file paths; secret groups "{value}", "{value}", ... all have output file "{value}".<br><br>SP fails to start | | |
+
+#### Integration / E2E tests
+
+#### Security testing
+
+Security testing will include:
+
+- Automated Vulnerability scans
+
+####  Performance testing
+
+As described in the [Performance](#performance) section above, the performance
+of the Secrets Provider for Push-to-File mode will be manually measured using
+the following metrics:
+
+- Latency (processing time) for parsing Pod annotations
+- Latency for generating secrets requests and processing responses
+- Latency for writing retrieved secrets to files
+
+_**QUESTION: Do we need to measure CPU and memory footprints?**_
+
+The measurements will be made using a representative configuration in terms
+of the number of secrets groups and the number of secrets per secrets group.
+Details are TBD, but a representative configuration might look something like
+this:
+
+- 5 secrets groups
+- 5 secrets / secrets group
+- Mix of secret value lengths (10 to 20 characters, plus some base64-encoded
+  TLS certificates)
+
+## Logs
```

Comment:
Can we reference the "Error Handling / Recovery / Supportability tests" for the error modes, or should we combine these two tables?

Reply:
Agreed.  I assume there are tests for all the error codes?